### PR TITLE
[FLINK-22471][flink-table][flink-connectors] Improvements to ConfigOption descriptions

### PR DESCRIPTION
**Pull request by Airblader**
## What is the purpose of the change

This improves some of the `ConfigOption`s' descriptions in various connectors by
* Fixing minor grammatical issues (spelling, interpunctuation, …)
* Removing default values from the description
* Using a proper `Description` with list block elements etc.

It's not exhaustive yet, and unfortunately for some things I wasn't sure if I understood the current description correctly, and if so tried to keep original phrasing as to not reflect anything incorrectly.

## Brief change log

- Updates to descriptions of connector options for Kinesis, Kafka, JDBC, ES and DataGen.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java [hunk 1]
```diff
@@ -166,9 +174,16 @@ private KafkaOptions() {}
...
                     .defaultValue("group-offsets")
                     .withDescription(
-                            "Optional startup mode for Kafka consumer, valid enumerations are "
-                                    + "\"earliest-offset\", \"latest-offset\", \"group-offsets\", \"timestamp\"\n"
-                                    + "or \"specific-offsets\"");
+                            Description.builder()
+                                    .text(
+                                            "Optional startup mode for Kafka consumer, valid enumerations are")
+                                    .list(
+                                            text("earliest-offset"),
```
Comment (twalthr) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java:181]: maybe still surround it by `'...'`

File: flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java [hunk 2]
```diff
@@ -200,45 +215,69 @@ private KafkaOptions() {}
...
-                                    + "\"default\": (use kafka default partitioner to partition records),\n"
-                                    + "\"fixed\": (each Flink partition ends up in at most one Kafka partition),\n"
-                                    + "\"round-robin\": (a Flink partition is distributed to Kafka partitions round-robin when 'key.fields' is not specified.)\n"
-                                    + "\"custom class name\": (use a custom FlinkKafkaPartitioner subclass)");
+                            Description.builder()
+                                    .text(
+                                            "Optional output partitioning from Flink's partitions into Kafka's partitions. Valid enumerations are")
+                                    .list(
+                                            text(
+                                                    "default (use kafka default partitioner to partition records)"),
```
Comment (twalthr) [inline flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaOptions.java:223]: maybe still surround it by '...' here. and below

**Review by twalthr — CHANGES_REQUESTED** [review]
Thanks @Airblader for cleaning this up. I added some minor comments. Esp. we should still highlight values with `'...'`.

**Comment by Airblader** [comment]
@twalthr Thanks, I've updated the PR!

**Comment by twalthr** [comment]
Thanks @Airblader. Merging...

**Review by twalthr — APPROVED** [review]
Thanks @Airblader. Merging...

**Pull request closed without merging**